plugins/woocommerce/includes/bis/class-wc-bis-sync.php

```diff
@@ -0,0 +1,274 @@
+<?php
+/**
+ * WC_BIS_Sync class
+ *
+ * @package  WooCommerce Back In Stock Notifications
+ * @since    1.0.0
+ */
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Sync Stock Controller.
+ *
+ * @class    WC_BIS_Sync
+ * @version  x.x.x
+ */
+class WC_BIS_Sync {
+
+	/**
+	 * Mark sync.
+	 *
+	 * @var boolean
+	 */
+	private $sync_needed = false;
+
+	/**
+	 * Product ids that need syncing at shutdown.
+	 *
+	 * @var array
+	 */
+	private $queue = array();
+
+	/**
+	 * Init.
+	 */
+	public function __construct() {
+
+		// Stock status changes.
+		add_action( 'woocommerce_product_set_stock_status', array( $this, 'product_stock_status_changed' ), 100, 3 );
+		add_action( 'woocommerce_variation_set_stock_status', array( $this, 'product_stock_status_changed' ), 100, 3 );
+
+		// Stock amount changes.
+		add_action( 'woocommerce_product_set_stock', array( $this, 'product_stock_changed' ), 100 );
+		add_action( 'woocommerce_variation_set_stock', array( $this, 'product_stock_changed' ), 100 );
+
+		// Save variable product and get the synced variations statuses.
+		add_action( 'shutdown', array( $this, 'sync' ), 100 );
+	}
+
+	/**
+	 * Sync stock.
+	 *
+	 * @return void
+	 */
+	public function sync() {
+
+		if ( ! $this->sync_needed ) {
+			return;
+		}
+
+		if ( wc_bis_debug_enabled() ) {
+			WC_BIS()->log( 'Product IDs: ' . print_r( $this->queue, true ), 'info', 'wc_bis_sync_logs' );
```

Comment:
It would be nice to just use `wc_get_logger()` directly here, and also clean this up to use [logging best practices](https://developer.woocommerce.com/docs/logging-in-woocommerce/#3-logging-best-practices) (i.e. don't output the array into the log message).

Reply:
Makes sense, thanks. Updated in [6508003](https://github.com/woocommerce/woocommerce/pull/55787/commits/6508003c3d0f1fc10032af4249040b53cfeeb0e0).